### PR TITLE
Lecture2 task2

### PR DESCRIPTION
**Pull request by mexal17**
Задание №2. 
Студенты на экзамене.
First refactoring after mentor's comments.

### CODE REVIEW

File: Lecture-2/Task2/l2task2/Group.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    private Student[] students;
+    private final int groupId;
+
+    public Group(int groupId, int studentAmount) {
+        this.groupId = groupId;
+        createStudents(studentAmount);
+    }
+
+    public void createStudents(int studentAmount) {
```
Comment (ivan-pronin) [inline Lecture-2/Task2/l2task2/Group.java]: Creating students is not responsibility of this class. You should pass into `group` already created students array

File: Lecture-2/Task2/l2task2/Application.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+    private static int bestScoreInGroup(int[] scores) {
+        int bestScore = 1;
+        for (int score : scores) {
+            bestScore = Math.max(score, bestScore);
+        }
+        return bestScore;
+    }
+
+    private static void showStudentsByScores(int score, Exam exam) {
```
Comment (ivan-pronin) [inline Lecture-2/Task2/l2task2/Application.java:58]: It's still pretty difficult to figure out which student got which ticket. If somehow tickets ordering will be changed, all results will be lost
So, better approach is to create a class `ExamAnswer` with fields `Student, Ticket and Score`. `Exam` class in this case will hold references to `Group` and `List<ExamAnswer>`

**Review by ivan-pronin — CHANGES_REQUESTED** [review]

File: Lecture-2/Task2/l2task2/Group.java (hunk 1 above)
Comment (mexal17) [reply Lecture-2/Task2/l2task2/Group.java]: done

File: Lecture-2/Task2/l2task2/Application.java (hunk 2 above)
Comment (mexal17) [reply Lecture-2/Task2/l2task2/Application.java:58]: done

File: Lecture-2/Task2/l2task2/Application.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
+package l2task2;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ThreadLocalRandom;
+
+public class Application {
+
+    private static ThreadLocalRandom RANDOM = ThreadLocalRandom.current();
```
Comment (ivan-pronin) [inline Lecture-2/Task2/l2task2/Application.java]: final

**Review by ivan-pronin — APPROVED** [review]

**Pull request merged**